File: dbflow-core/src/main/java/com/raizlabs/android/dbflow/annotation/NotNull.java
package com.raizlabs.android.dbflow.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Description: Specifies that a {@link Column} is not null.
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.SOURCE)
public @interface NotNull {

    /**
     * Defines how to handle conflicts for not null column
     *
     * @return a {@link com.raizlabs.android.dbflow.annotation.ConflictAction} enum
     */
    ConflictAction onNullConflict() default ConflictAction.FAIL;

}

File: dbflow-core/src/main/java/com/raizlabs/android/dbflow/annotation/NotNull.kt
package com.raizlabs.android.dbflow.annotation

/**
 * Description: Specifies that a [Column] is not null.
 */
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.SOURCE)
annotation class NotNull(
        /**
         * Defines how to handle conflicts for not null column
         *
         * @return a [com.raizlabs.android.dbflow.annotation.ConflictAction] enum
         */
        val onNullConflict: ConflictAction = ConflictAction.FAIL)
